src/events/events.ts

```diff
@@ -34,40 +34,53 @@ import {
     genMemberUpdateEmbed, genMessageDeleteEmbed, genMessageEditEmbed, genUserUpdateEmbed
 } from "./logs";
 import { Blacklist } from "../schemas/blacklist";
+import { formatGoLivePost } from "./shoutout";
 
 /*
  * This module has event listeners for the bot
  */
 
 /**
- * Logs member join events to servers subscribed to logs.
+ * Logs an embed to a log channel if possible. Does nothing if the
+ * server doesn't have logs configured. Deletes configuration if logs isn't a channel.
  * 
  * @param client the Discord bot
- * @param member the member who joined
+ * @param server the server document from the database
+ * @param embed the embed to log
  */
-export const onMemberJoin = async (client: Client, member: GuildMember): Promise<void> => {
-    // The server and logs channel ID of the server that the member joined
-    const server: Server = await Server.get(member.guild.id);
+const logTo = async (client: Client, server: Server, embed: EmbedBuilder): Promise<void> => {
     const logsID: string = server.logs;
 
     // Skip if the server isn't subscribed to logs
     if (!logsID) {
         return;
     }
 
+    // Logs is guaranteed to be a logs channel due to the constraints
+    // of the /config logs command
     const logs: TextChannel = <TextChannel> client.channels.cache.get(logsID);
 
-    try {
-        // Log to the logs channel
-        await logs.send({ embeds: [genMemberJoinEmbed(member)] });
-    } catch {
-        // Remove configurations if logging failed.
-        // Most likely causes are that the channel was deleted or that
-        // Sushi Bot had its permissions revoked
+    if (!logs) {
+        // Remove configurations if the logs do not exist
         server.logs = null;
         await server.save();
     }
```

Comment:
The logic for logging and embed to a channel has been moved to this helper function.